### PR TITLE
test(frontend): raise coverage and ratchet floor

### LINKED ISSUE #509: test(frontend): increase statement coverage from 57% to 70%+
**Parent:** #503
**Priority:** 🟠 Strategic

**Affected:** `src/routes/salaries/+page.svelte` (30.71% coverage), `src/routes/simulations/+page.svelte` (37.99%), `e2e/` E2E test suite

**Why:** Frontend statement coverage is 57.28% overall; several pages have < 40% branch coverage. High-traffic routes (salaries, simulations, dashboard) lack edge-case test coverage. CI uploads coverage but has no enforcement floor, allowing regressions to ship.

**Impact:** High - silent failures in dashboard, snapshot, and financial simulation flows can reach users.
**Contagion:** High - every page touch adds untested code paths.
**Effort:** Days - add unit tests for page logic, e2e tests for interaction flows.
**Business Alignment:** High - protects core finance workflows.

**Fix:**
1. Set a coverage floor in `vitest.config.ts`: e.g., 70% statements, 60% branches (ratchet upward over time).
2. Add unit tests for high-value page logic: salary form validation, snapshot creation flows, simulation state management.
3. Add e2e tests for critical happy paths: "create salary" → "see chart update", "save snapshot" → "list includes new".
4. Run `npm test -- --run --coverage` locally before PR; CI will enforce the floor.

**Test approach:** Run `npm test:coverage` and verify report shows ≥70% statements on touched modules. Run `npm run test:e2e` for integration checks.

**Estimated:** 2-3 days

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

The tech-debt audit flagged frontend statement coverage at 57.28%, with `src/routes/salaries/+page.svelte` at 30.71% and `src/routes/simulations/+page.svelte` at 37.99%. The original goal was 70% statements / 60% branches; getting all the way there is a multi-day push that mostly depends on chipping at the 2,400-line salaries page.

This PR moves the needle and sets a higher coverage floor so future PRs can't regress what we just gained.

## Implementation information

New / extended tests:

- `src/lib/utils/charts/simulations.test.ts`: cover the tooltip formatter on both the array and single-param paths, the y-axis label formatter, and the palette wrap-around (more series than colours).
- `src/routes/simulations/page.test.ts`: render-only test extended with a real run-simulation happy path (POST body shape + results section appears) and a non-2xx error path.
- `src/routes/salaries/page.test.ts`: add a bonus-modal POST flow plus the amount-must-be-positive validation gate.
- `vite.config.ts`: thresholds raised from 56/39/56/59 to 62/43/60/64 so the new floor is enforced.

Coverage moves from 57.28% / 41.22% / 56.28% / 59.80% to **63.61% / 43.96% / 62.86% / 65.40%** (statements / branches / functions / lines).

### Open follow-ups

The original 70% / 60% target isn't fully reached here. Most of the remaining gap is in `salaries/+page.svelte` (now 38%, four big modals still mostly untested). Refactoring that file (the audit's #508) will make those flows easier to cover incrementally.

Closes #509

> Changelog: test(frontend): raise coverage and ratchet floor

### CODE REVIEW

File: src/routes/salaries/page.test.ts [hunk 1]
```diff
@@ -203,6 +203,55 @@ describe('Salaries page — saveSalary validation & error display', () => {
...
+		expect((init as RequestInit).method).toBe('POST');
+		const body = JSON.parse((init as RequestInit).body as string);
+		expect(body).toMatchObject({
+			date: '2026-05-01',
+			amount: 12000,
+			currency: 'PLN',
+			type: 'annual',
+			company: 'Acme',
+			contract_type: 'UOP'
+		});
```
[the remark this answers is not in the document]
Comment (Automaat) [reply src/routes/salaries/page.test.ts:234]: Added an assertion that `Data wypłaty` (a field only present inside the bonus modal) is no longer in the DOM after the successful save.
